migrate/migrate_repos.py

```diff
@@ -105,5 +231,128 @@ def extract_repo_names(repo_list_file):
+                click.secho(f"  {team}", bold=True, nl=False)
+                click.echo(f" used in {', '.join(repos_team_is_used_in)}")
+
+        if missing_usernames_to_repos:
+            click.echo("Missing Users:")
+            for username in sorted(missing_usernames_to_repos):
+                repos_user_is_used_in = sorted(missing_usernames_to_repos[username])
+                click.secho(f"  {username}", bold=True, nl=False)
+                click.echo(f" used in {', '.join(repos_user_is_used_in)}")
+
+        if skip_missing_teams:
+            click.secho("Ignoring missing teams and users!", bold=True)
+            # Return a modified version of repos_to_permissions that has the
+            # missing information stripped out.
+            stripped_repos_to_permissions = {}
+            for repo, permissions in repos_to_permissions.items():
+                stripped_repos_to_permissions[repo] = dataclasses.replace(
+                    permissions,
+                    teams={team: role for team, role in permissions.teams.items() if team in known_team_slugs},
+                    users={username: role for username, role in permissions.users.items() if username in known_usernames}
+                )
+            return stripped_repos_to_permissions
+        else:
+            sys.exit(
+                "Exiting because teams or users are missing. "
+                "To ignore this check, use --skip-missing-teams"
+            )
+
+    return repos_to_permissions
+
+
+def set_repo_permissions(permissions, api, dest_org, preview):
+    """
+    Set team permissions for one repo.
+
+    Won't actually make any GitHub API calls if preview is True.
+
+    The permissions param is a dict of repo slugs to RepoPermissions objects.
+
+    This function only sets team permissions and not user permissions. User
+    roles will follow over with the transferred repository–there's no need to
+    set them manually here.
+    """
+    # teams_by_role is a mapping of access role (e.g. "admin") to a list of
+    # team slugs.
+    teams_by_role = defaultdict(list)
+    for team_slug, access in sorted(permissions.teams.items()):
+        teams_by_role[access].append(team_slug)
+
+    # The permissions update call requires that the owner be specified, but the
+    # owner for a repo in an org is just the user that has the login of the org
+    # name (verified by looking through the results of getting repo listings
+    # from this API). So everything in the org "openedx" is owned by user
+    # "openedx".
+    owner = dest_org
+
+    for access, teams in sorted(teams_by_role.items()):
+        click.secho(f"     assigning {access} teams:", italic=True)
+        click.echo("      ", nl=False)
+        for team_slug in teams:
+            click.echo(f" {team_slug}", nl=False)
+            if not preview:
+                api.teams.add_or_update_repo_permissions_in_org(
+                    dest_org, team_slug, owner, permissions.slug, access
```

Comment:
nit, optional: I think this function call would be easier to read and verify if you passed these args in as kwargs.

Reply:
Sure thing, will update.